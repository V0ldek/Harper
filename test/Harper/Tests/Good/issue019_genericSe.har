app :: (a -> b) -> a -> b;
app f x = f x;

se :: Integer -> sideeffect -> Integer;
se n = {
  eval printLn "Sideeffect." ();
  return n;
};

main :: sideeffect -> ();
main = {
  eval printLn (app (se 42) ()) ();
};