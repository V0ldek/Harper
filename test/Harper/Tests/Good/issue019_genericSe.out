Sideeffect.
42

Execution ended with value:
()